### PR TITLE
desktop_entry: include high resolution icons

### PR DESCRIPTION
**Pull request by sevu**
I placed the new images in `packaging` and removed  the `icons` folder.
In case of scons the whole directory is copied, so the .desktop file had to go somewhere else.
16x16 is not using a downscaled icon, but a pixelart icon, which is wesnoth.org's favicon for this size.

It's temping to rename the icon to just wesnoth.png, however, this would have an effect on existing icon-themes on some distributions (which call the icon exactly like that, e.g. arch, but not debian/ubuntu)

### CODE REVIEW

File: SConstruct [hunk 1]
```diff
@@ -638,11 +638,8 @@ if env["nls"]:
     env.InstallData("datadir", "wesnoth", "l10n-track")
 InstallManpages(env, "wesnoth")
 if have_client_prereqs and have_X and env["desktop_entry"]:
-     if sys.platform == "darwin":
-         env.InstallData("icondir", "wesnoth", "icons/wesnoth-icon-Mac.png")
-     else:
-         env.InstallData("icondir", "wesnoth", "icons/wesnoth-icon.png")
-     env.InstallData("desktopdir", "wesnoth", "icons/wesnoth.desktop")
+     env.InstallData("icondir", "wesnoth", "packaging/icons")
+     env.InstallData("desktopdir", "wesnoth", "packaging/wesnoth.desktop")
```
Comment (sevu) [inline SConstruct]: While cmake copies explicitly the hicolor folder, scons copies the contents of the icons folder, which ends up being the same.

**Comment by CelticMinstrel** [comment]
There's a typo in the commit message.

**Comment by sevu** [comment]
Thanks, fixed

**Comment by Vultraz** [comment]
Are you sure you didn't re-save these? They look rather grainy in the previews...

**Comment by sevu** [comment]
If this images are just a downscaled pngs I guess it would be the same as supplying one highdpi image and let the Desktop Environment downscale it (the 16x16 being an exception). But I'm not an expert on that.

**Comment by sevu** [comment]
I consider closing this and just adding the 128x128, for simplicities sake.
128x128 because I saw that there exists an image in that dimension for the map editors logo.
@rhonda Do you know if the approach in this PR has advantages over just using one image file (like it's currently done)?

**Comment by GregoryLundberg** [comment]
@sevu We want to avoid a desktop upscaling at all costs. The reason providing the full set of recommended scales is A Good Thing is that no DE downscales with the quality of the full-fledged graphics systems like The GIMP and Photoshop. Other than image quality, the DE authors recommend providing the pre-scaled images for their own performance reasons.

My recommendation would be to provide the full set of sizes for Windows and MacOS to give the packagers the best quality; and make it clear to them they are free to pick-and-choose for their specific targets.

**Comment by sevu** [comment]
The icons sizes which would be needed would be as follows:

16x16 (is here, pixelart instead of downscalled image)
22x22
24x24 (could be the 22x22 one with an 1px transparent frame)
32x32 *(is here)*
36x36
48x48
62x62 *(is here)*
72x72
96x96
128x128 *(is here)*
192x192
256x256 *(is here)*
384x384
480x480
512x512 *(is here)*

For now I would merge this, if nobody objects. The other icons can be added later if someone creates the files. ( I need an icon >= 128x128 for adding an appdata file, so it would be good to have this merged)

**Comment by GregoryLundberg** [comment]
Please fix the merge conflict.

Then, I agree, this should merge unless someone jumps in to object before you're ready to click the big green button.

**Pull request merged**